go-getting-started/main.go

```diff
@@ -54,112 +59,242 @@ func (j jokesHandler) parseSkipAndLimit(w http.ResponseWriter, r *http.Request)
 		w.WriteHeader(http.StatusBadRequest)
 		return Pagination{}, nil
 	}
-	if skip == limit {
-		w.WriteHeader(http.StatusBadRequest)
-	}
+
 	pagination := Pagination{Skip: skip, Limit: limit}
 	return pagination, nil
 }
 
-func (p jokesHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+func (j jokesHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 
 	if r.Method == http.MethodPost {
 		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
 		return
 	}
 
-	if len(p.jokes) == 0 {
+	if len(j.jokes) == 0 {
 		w.WriteHeader(http.StatusNoContent)
-		json.NewEncoder(w).Encode(p.jokes)
+		err := json.NewEncoder(w).Encode(j.jokes)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
 		return
 	}
 
-	if len(p.jokes) == 1 {
-		json.NewEncoder(w).Encode(p.jokes)
+	if len(j.jokes) == 1 {
+		err := json.NewEncoder(w).Encode(j.jokes)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
 		return
 	}
 
-	jokes := p.jokes
+	jokes := j.jokes
 
-	pagination, err := p.parseSkipAndLimit(w, r)
+	pagination, err := j.parseSkipAndLimit(w, r)
 
 	if err != nil {
 		fmt.Fprintln(w, err)
 	}
 
 	res := jokes[pagination.Skip : pagination.Limit+pagination.Skip]
 
-	json.NewEncoder(w).Encode(res)
+	err = json.NewEncoder(w).Encode(res)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+func (j jokesHandler) getJokes(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("content-type", "application/json")
+
+	var jokes []models.Joke
+
+	pagination, err := j.parseSkipAndLimit(w, r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+	}
+
+	findOptions := options.Find()
+	findOptions.SetLimit(int64(pagination.Limit)).SetSkip(int64(pagination.Skip))
+
+	cursor, err := j.collection.Find(context.TODO(), bson.M{}, findOptions)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	defer cursor.Close(context.TODO())
+	for cursor.Next(context.TODO()) {
+		var joke models.Joke
+		err := cursor.Decode(&joke)
+		if err != nil {
+			panic(err)
+		}
+		jokes = append(jokes, joke)
+	}
+
+	err = json.NewEncoder(w).Encode(jokes)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+func (j jokesHandler) getId(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("content-type", "application/json")
+
+	_, err := j.collection.Indexes().CreateOne(
```

Comment:
Creating indexed has nothing to do with processing a HTTP request.
This code should exist in the storage layer, and be only invoked on service start.